napari_btrack/track.py

```diff
@@ -31,23 +151,276 @@ def run_tracker(objects, config_file_path):
         return data, properties, graph
 
 
-def track() -> FunctionGui:
-    @magicgui(
-        call_button=True,
-        persist=True,
-        config_file_path=dict(value=Path.home()),
-        reset_button=dict(widget_type="PushButton", text="Reset defaults"),
+def get_save_path():
+    """Helper function to open a save configuration file dialog."""
+    show_file_dialog = use_app().get_obj("show_file_dialog")
+    save_path = show_file_dialog(
+        mode=FileDialogMode.OPTIONAL_FILE,
+        caption="Specify file to save btrack configuration",
+        start_path=None,
+        filter="*.json",
     )
-    def widget(
-        viewer: napari.Viewer,
-        segmentation: napari.layers.Image,
-        config_file_path: Optional[Path],
-        reset_button,
+    return save_path
+
+
+def get_load_path():
+    """Helper function to open a load configuration file dialog."""
+    show_file_dialog = use_app().get_obj("show_file_dialog")
+    load_path = show_file_dialog(
+        mode=FileDialogMode.EXISTING_FILE,
+        caption="Choose JSON file containing btrack configuration",
+        start_path=None,
+        filter="*.json",
+    )
+    return load_path
+
+
+def html_label_widget(label: str, tag: str = "b") -> dict:
+    """
+    Create a HMTL label widget.
+    """
+    return dict(
+        widget_type="Label",
+        label=f"<{tag}>{label}</{tag}>",
+    )
+
+
+def _create_per_model_widgets(model: BaseModel) -> List[Widget]:
+    """
+    For a given model create the required list of widgets.
+    The items "hypotheses" and the various matrices need customisation,
+    otherwise we can use the napari default.
+    """
+    widgets: List[Widget] = []
+    if model:
```

Comment:
Given the typing of `model` it can't be `None` and therefore this `if` is redundant?

Reply:
We were passing None at runtime occasionally so it was still needed (both default tracker configs have an attribute called `object_model` which is `None`, but I've decided the better solution is to not pass `None` here...